pkg/controller/cluster/cluster.go

```diff
@@ -86,15 +141,34 @@ func (c *ClusterReconciler) Reconcile(ctx context.Context, req reconcile.Request
 		}
 
 		klog.Infof("enqueue cluster [%s]", cluster.Name)
+		if err := c.createCluster(ctx, &cluster); err != nil {
+			return reconcile.Result{}, util.LogAndReturnErr("failed to create cluster", err)
+		}
+		return reconcile.Result{}, nil
+	}
+
+	// remove finalizer from the server pods and update them.
+	matchingLabels := client.MatchingLabels(map[string]string{"role": "server"})
+	listOpts := &client.ListOptions{Namespace: util.ClusterNamespace(&cluster)}
+	matchingLabels.ApplyToList(listOpts)
 
-		return reconcile.Result{}, c.createCluster(ctx, &cluster)
+	if err := c.Client.List(ctx, &podList, listOpts); err != nil {
+		return reconcile.Result{}, client.IgnoreNotFound(err)
+	}
+	for _, pod := range podList.Items {
+		if controllerutil.ContainsFinalizer(&pod, etcdPodFinalizerName) {
+			controllerutil.RemoveFinalizer(&pod, etcdPodFinalizerName)
```

Comment:
Is there an error check supposed to happen here?

Reply:
no not really, the removeFinalizer and AddFinalizer both return a boolean that will only suggests if the finalizer has being deleted, but it will be redundant since I added ContainsFinalizer() check